app/uk/gov/hmrc/cgtpropertydisposals/service/TaxEnrolmentService.scala

```diff
@@ -103,7 +149,30 @@ class TaxEnrolmentServiceImpl @Inject()(
                                 .leftMap(
                                   error => Error(s"Could not check database to determine subscription status: $error")
                                 )
-      _ <- EitherT.liftF(handleDatabaseResult(maybeEnrolmentRequest))
+      maybeUpdateVerifierRequest <- verifiersRepository
+                                     .get(ggCredId)
+                                     .leftMap(
+                                       error =>
+                                         Error(
+                                           s"Could not check database to determine update verifier request exists : $error"
+                                       )
+                                     )
+      _ <- EitherT.liftF(handleEnrolmentState(maybeEnrolmentRequest -> maybeUpdateVerifierRequest))
     } yield maybeEnrolmentRequest
 
+  @SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
```

Comment:
do you need this `SupressWarnings`?

Reply:
Yes.

```scala
error] /home/mikail/Projects/hmrc/cgt-property-disposals/app/uk/gov/hmrc/cgtpropertydisposals/service/TaxEnrolmentService.scala:175: [wartremover:NonUnitStatements] Statements must return Unit
[error]       EitherT.rightT[Future, Error](Right(()))
```

Reply:
oh, it's because you don't need the `Right` in the `rightT`

Reply:
:man_facepalming:

Reply:
it shouldn't have compiled